File: eval/code_corpus/vulnerable/command_injection.js
// # INTENTIONALLY VULNERABLE — eval fixture, do not deploy
// OS command injection — user input passed to a shell via child_process.exec.
// Ground-truth finding: vuln_class=command-injection on the exec line below.

const { exec } = require("child_process");

function ping(host, cb) {
  // VULN: exec runs the string in a shell — host can inject `; rm -rf /`.
  exec("ping -c 1 " + host, cb);
}

module.exports = { ping };

File: eval/code_corpus/vulnerable/command_injection.py
# INTENTIONALLY VULNERABLE — eval fixture, do not deploy
"""OS command injection — user input passed to a shell.

Ground-truth finding: vuln_class=command-injection on the subprocess line below.
"""
import subprocess


def ping(host: str):
    # VULN: shell=True with interpolated input — command injection sink.
    return subprocess.check_output("ping -c 1 " + host, shell=True)
